src/components/EditWebhooks.js

```diff
@@ -0,0 +1,259 @@
+import { Button, Form } from 'react-bootstrap';
+import { defineMessages } from 'react-intl';
+import { get, pick, isEmpty } from 'lodash';
+import { compose, graphql } from 'react-apollo';
+import gql from 'graphql-tag';
+
+import withIntl from '../lib/withIntl';
+import InputField from './InputField';
+import events from '../constants/notificationEvents';
+import Loading from './Loading';
+
+class EditWebhooks extends React.Component {
+  static propTypes = {
+    collectiveSlug: PropTypes.string.isRequired,
+    /** From graphql query */
+    data: PropTypes.object.isRequired,
+    /** From intl */
+    intl: PropTypes.object.isRequired,
+  };
+
+  static getDerivedStateFromProps(props, state) {
+    const webhooks = {};
+
+    if (isEmpty(state.webhooks) && !isEmpty(props.data.Collective)) {
+      get(props, 'data.Collective.notifications', []).forEach(x => {
+        if (!(x.webhookUrl in webhooks)) {
+          webhooks[x.webhookUrl] = pick(x, ['webhookUrl']);
+          webhooks[x.webhookUrl].activities = [];
+        }
+        webhooks[x.webhookUrl].activities.push(x.type);
+      });
+
+      return { ...state, webhooks: Object.values(webhooks) };
+    }
+
+    return state;
+  }
+
+  constructor(props) {
+    super(props);
+    const { intl } = props;
+
+    this.state = {
+      modified: false,
+      webhooks: {},
+      status: null,
+    };
+
+    this.messages = defineMessages({
+      'webhooks.url.label': {
+        id: 'webhooks.url.label',
+        defaultMessage: 'URL',
+      },
+      'webhooks.types.label': {
+        id: 'webhooks.types.label',
+        defaultMessage: 'Activities',
+      },
+      'webhooks.add': {
+        id: 'webhooks.add',
+        defaultMessage: 'add another webhook',
+      },
+      'webhooks.remove': {
+        id: 'webhooks.remove',
+        defaultMessage: 'remove webhook',
+      },
+      loading: { id: 'loading', defaultMessage: 'loading' },
+      save: { id: 'save', defaultMessage: 'save' },
+      saved: { id: 'saved', defaultMessage: 'saved' },
+    });
+
+    this.fields = [
+      {
+        name: 'webhookUrl',
+        maxLength: 255,
+        type: 'url',
+        label: intl.formatMessage(this.messages['webhooks.url.label']),
+        required: true,
+      },
+      {
+        name: 'activities',
+        type: 'select',
+        label: intl.formatMessage(this.messages['webhooks.types.label']),
+        options: events.sort().map(event => {
+          return { value: event, label: event };
+        }),
+        multiple: true,
+        defaultValue: [],
+        required: true,
+      },
+    ];
+  }
+
+  editWebhook = (index, fieldname, value) => {
+    const { webhooks } = this.state;
+    webhooks[index][fieldname] = value;
+    this.setState({ webhooks, modified: true });
+  };
+
+  addWebhook = webhook => {
+    const { webhooks } = this.state;
+    webhooks.push(webhook || {});
+    this.setState({ webhooks, modified: true });
+  };
+
+  removeWebhook = index => {
```

Comment:
Remove doesn't work

![Peek 09-05-2019 18-39](https://user-images.githubusercontent.com/1556356/57470711-c7d19d00-7289-11e9-9989-f92cd7bbc930.gif)

Reply:
Fixed.

Reply:
Should work fine now.